harness/determined/deploy/local/cluster_utils.py

```diff
@@ -177,56 +185,167 @@ def master_up(
         )
         make_temp_conf = True
 
-    # Ensure checkpoint storage directory exists.
-    final_storage_host_path = master_conf.get("checkpoint_storage", {}).get("host_path")
-    if final_storage_host_path is not None:
-        final_storage_host_path = Path(final_storage_host_path)
-        if not final_storage_host_path.exists():
-            final_storage_host_path.mkdir(parents=True)
-
     if make_temp_conf:
         fd, temp_path = tempfile.mkstemp(prefix="det-deploy-local-master-config-")
         with open(fd, "w") as f:
             yaml.dump(master_conf, f)
-        master_config_path = Path(temp_path)
+        master_config_path = pathlib.Path(temp_path)
 
     # This is always true by now, but mypy needs help.
     assert master_config_path is not None
+    restart_policy = "unless-stopped" if autorestart else "no"
+    env = {
+        "DET_MASTER_HTTP_PORT": str(port),
+        "DET_DB_PASSWORD": db_password,
+        "DET_LOG_INFO": "info",
+    }
 
-    env["DET_MASTER_CONFIG"] = str(master_config_path.resolve())
+    # Kill existing master container if exists.
+    master_down(master_name=master_name, delete_db=delete_db, cluster_name=cluster_name)
+
+    # Create network.
+    client = docker_client()
+
+    # Start db and wait for healthcheck.
+    db_name = f"{cluster_name}_{DB_NAME}"
+    volume_name = f"{cluster_name}_{VOLUME_NAME}"
+
+    @contextlib.contextmanager
+    def defer_cleanup(fn: Callable, *args: Any) -> Generator:
```

Comment:
I think you could:
```
def defer_cleanup(fn: Callable[[], None]) -> Generator:
```

then use `lambda`s:

```
exit_stack.enter_context(defer_cleanup(lambda: remove_network(NETWORK_NAME)))
```

And that would give you mypy type-checking on the deferred functions, which would be nice.  No point in using `Any` when it's not necessary, it just makes it more likely for a bug to appear either now or in the future.

There are other, more elaborate ways to do it, but I think that's the simplest and easiest.